### PR TITLE
[Comp] Patch missing type shapes in a fly 

### PR DESCRIPTION
**Pull request by dkijania**
When there is a missing reference file we can checkout missing commit , dump type shape and continue compairsion

### CODE REVIEW

**Comment by dkijania** [comment]
!ci-build-me

**Comment by dkijania** [comment]
!ci-build-me

**Comment by dkijania** [comment]
!ci-build-me

**Comment by dkijania** [comment]
!ci-build-me

**Comment by dkijania** [comment]
!ci-build-me

**Comment by dkijania** [comment]
!ci-build-me

**Comment by dkijania** [comment]
!ci-build-me

**Comment by dkijania** [comment]
!ci-build-me

**Comment by dkijania** [comment]
!ci-build-me

**Comment by dkijania** [comment]
!ci-build-me

**Comment by dkijania** [comment]
!ci-build-me

**Comment by dkijania** [comment]
!ci-build-me

**Comment by dkijania** [comment]
!ci-build-me

**Comment by dkijania** [comment]
!ci-build-me

**Comment by dkijania** [comment]
!ci-build-me

**Comment by dkijania** [comment]
!ci-build-me

**Comment by dkijania** [comment]
!ci-build-me

**Comment by dkijania** [comment]
!ci-build-me

**Comment by dkijania** [comment]
!ci-build-me

**Review by SanabriaRusso — APPROVED** [review]

**Comment by georgeee** [comment]
@dkijania seems like this branch is a bit broken, I see some commits seemingly targeted at `develop`, however the base branch is `compatible`.

File: README.md [hunk 1]
```diff
@@ -1,14 +1,16 @@
 ### Build status
 
-| Develop | Berkeley | Compatible | 
-| ------- | -------- | ---------- |
-| [![Build status - develop](https://badge.buildkite.com/0c47452f3ea619d3217d388e0de522b218db28c3e161887a9a.svg?branch=develop)](https://buildkite.com/o-1-labs-2/mina-end-to-end-nightlies) | [![Build status - berkeley](https://badge.buildkite.com/0c47452f3ea619d3217d388e0de522b218db28c3e161887a9a.svg?branch=berkeley)](https://buildkite.com/o-1-labs-2/mina-end-to-end-nightlies) | [![Build status - compatible](https://badge.buildkite.com/0c47452f3ea619d3217d388e0de522b218db28c3e161887a9a.svg?branch=compatible)](https://buildkite.com/o-1-labs-2/mina-end-to-end-nightlies)
+| Develop | Compatible | Master |
```
Comment (mrmr1993) [inline README.md:3]: This (and other) changes shoudn't be here. Please remove them.

**Review by mrmr1993 — CHANGES_REQUESTED** [review]

**Comment by dkijania** [comment]
!ci-build-me

**Comment by dkijania** [comment]
!ci-build-me

**Comment by dkijania** [comment]
!ci-build-me

**Comment by dkijania** [comment]
!ci-build-me

**Comment by dkijania** [comment]
!ci-build-me

File: buildkite/scripts/version-linter-patch-missing-type-shapes.sh [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    git submodule update --init --recursive
+    eval $(opam config env)
+    TYPE_SHAPE_FILE=${__commit:0:7}-type_shape.txt
+    dune exec src/app/cli/src/mina.exe internal dump-type-shapes > /tmp/${TYPE_SHAPE_FILE}
+    revert_checkout
+    source buildkite/scripts/gsutil-upload.sh /tmp/${TYPE_SHAPE_FILE} gs://mina-type-shapes
+}
+
+if ! $(gsutil ls gs://mina-type-shapes/${BUILDKITE_COMMIT:0:7} 2>/dev/null); then
+    checkout_and_dump $BUILDKITE_COMMIT
```
[the remark this answers is not in the document]
Comment (dkijania) [reply buildkite/scripts/version-linter-patch-missing-type-shapes.sh:38]: Yes, this is only another gate that make sure there is no missing files as this one can be used as  base branch or release branch reference file. I will remove this if

**Comment by dkijania** [comment]
!ci-build-me

File: scripts/version-linter.py [hunk 3]
```diff
@@ -250,18 +269,18 @@ def assert_commit(commit, desc):
...
-  base_branch_commit=branch_commit(base_branch)
-  download_type_shapes('base',base_branch,base_branch_commit)
+  base_branch_commit = find_latest_type_shape_ref_on(base_branch,n=10)
+  download_type_shape('base',base_branch,base_branch_commit)
 
   print('')
 
-  release_branch_commit=branch_commit(release_branch)
-  download_type_shapes('release',release_branch,release_branch_commit)
+  release_branch_commit=find_latest_type_shape_ref_on(release_branch, n=10)
```
Comment (georgeee) [inline scripts/version-linter.py:277]: As I mentioned in `develop` twin of this PR, I'm not sure this 10-commit check is a good idea, or necessary.

**Comment by dkijania** [comment]
!ci-build-me

**Comment by dkijania** [comment]
!ci-build-me

**Comment by dkijania** [comment]
!ci-build-me

**Comment by dkijania** [comment]
!ci-build-me

**Comment by dkijania** [comment]
!ci-build-me

**Comment by dkijania** [comment]
!ci-build-me

**Comment by dkijania** [comment]
!ci-build-me

**Comment by dkijania** [comment]
!ci-build-me

**Comment by dkijania** [comment]
!ci-build-me

**Review by mrmr1993 — APPROVED** [review]

**Comment by mrmr1993** [comment]
!ci-build-me

**Pull request merged**